core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/UnaryPredicateOperator.java

```diff
@@ -82,4 +106,62 @@ private static FunctionResolver isNotNull() {
             .collect(
                 Collectors.toList()));
   }
+
+  private static FunctionResolver ifNull() {
+    FunctionName functionName = BuiltinFunctionName.IFNULL.getName();
+    List<ExprType> typeList = ExprCoreType.coreTypes();
+    typeList.add(UNKNOWN);
```

Comment:
Chen, we have to add UNKNOWN, otherwise, we will see the "UNKNOWN" type missing error, I think it is NONE type.
est_nullif_predicate()
com.amazon.opendistroforelasticsearch.sql.exception.ExpressionEvaluationException: nullif function expected {[BYTE,BYTE],[SHORT,SHORT],[INTEGER,INTEGER],[LONG,LONG],[FLOAT,FLOAT],[DOUBLE,DOUBLE],[BOOLEAN,BOOLEAN],[STRING,STRING],[TIMESTAMP,TIMESTAMP],[DATE,DATE],[TIME,TIME],[DATETIME,DATETIME],[INTERVAL,INTERVAL],[STRUCT,STRUCT],[ARRAY,ARRAY]}, but get [UNKNOWN,UNKNOWN]
	at com.amazon.opendistroforelasticsearch.sql.expression.function.FunctionResolver.resolve(FunctionResolver.java:46)
	at com.amazon.opendistroforelasticsearch.sql.expression.function.BuiltinFunctionRepository.resolve(BuiltinFunctionRepository.java:44)
	at com.amazon.opendistroforelasticsearch.sql.expression.function.BuiltinFunctionRepository.compile(BuiltinFunctionRepository.java:30)
	at com.amazon.opendistroforelasticsearch.sql.expression.DSL.function(DSL.java:500)
	at com.amazon.opendistroforelasticsearch.sql.expression.DSL.nullif(DSL.java:526)
	at com.amazon.opendistroforelasticsearch.sql.expression.operator.predicate.UnaryPredicateOperatorTest.test_nullif_predicate(UnaryPredicateOperatorTest.java:138)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)


test_isnull_predicate()
com.amazon.opendistroforelasticsearch.sql.exception.ExpressionEvaluationException: isnull function expected {[BYTE],[SHORT],[INTEGER],[LONG],[FLOAT],[DOUBLE],[BOOLEAN],[STRING],[TIMESTAMP],[DATE],[TIME],[DATETIME],[INTERVAL],[STRUCT],[ARRAY]}, but get [UNKNOWN]

Reply:
The null value doesn't have a type. the problem only related to NULL literal test, e.g. ifnull(NULL), do we need to support NULL literal?

Reply:
@harold-wang I think you can remove null literal test from your UT and code here. This null literal support is missing for all expressions. We need to handle it in one place.

Reply:
Let me remove null literal, thanks @penghuo @dai-chen

Reply:
@dai-chen remove the UNKNOWN(null) type will make isnull() function totally useless, how about to keep it for now and remove it once you can enable it in generic way, thx.

Reply:
@harold-wang I don't think so. If we don't consider NULL literal, you won't get any UNKNOWN type input. For example, in your IT, the value of 1/0 is null but its type is integer. Could you post an example in which test case this is required?

Reply:
All IT passed without UNKNOWN type, but UT has some coverage issue: what make sense to me:

  public static ExprValue exprIsNull(ExprValue v) {
    if (!v.isNull() && !v.isMissing()) {
      return LITERAL_FALSE;  // added by Harold, how can we cover here without UNKNOWN type.
    } else {
      return LITERAL_TRUE;
    }
  }

Reply:
UNKNOWN is removed now.